pyccel/parser/semantic.py

```diff
@@ -925,14 +927,27 @@ def _handle_kernel(self, expr, func, args, **settings):
                         severity='fatal')
             # TODO : type check the NUMBER OF BLOCKS 'numBlocks' and threads per block 'tpblock'
             if not isinstance(expr.numBlocks, LiteralInteger):
-                errors.report("Invalid Block number parameter for Kernel call",
+                try:
+                    numBlocks = self.scope.variables[expr.numBlocks]
+                    if not isinstance(numBlocks.dtype, NativeInteger):
+                        errors.report(INVALID_KERNEL_CALL_BP_GRID,
+                        symbol = expr,
+                        severity='error')
+                except KeyError:
+                    errors.report(INVALID_KERNEL_CALL_BP_GRID,
                         symbol = expr,
                         severity='error')
             if not isinstance(expr.tpblock, LiteralInteger):
-                errors.report("Invalid Thread per Block parameter for Kernel call",
+                try:
+                    tpblock = self.scope.variables[expr.tpblock]
```

Comment:
See above, this is not the right way to search for variables

Reply:
Fixed e5d7ba0ed13698404e6914ea4ea1b945f8d06c15